source/Government.cpp

```diff
@@ -140,6 +144,63 @@ void Government::Load(const DataNode &node)
 						grand.PrintTrace("Skipping unrecognized attribute:");
 				}
 		}
+		else if(key == "illegals")
+		{
+			if(!add)
+				illegals.clear();
+			for(const DataNode &grand : child)
+				if(grand.Size() >= 2)
+				{
+					bool remove = (grand.Token(0) == "remove");
+					if(remove || grand.Token(0) == "ignore")
+					{
+						const Outfit *outfit = GameData::Outfits().Get(grand.Token(1));
+						const auto &illegal = illegals.find(outfit);
+						if(illegal != illegals.end())
+						{
+							if(remove)
+								illegals.erase(illegal);
+							else
+								illegal->second = -1;
+						}
+						else if (!remove)
+							illegals.emplace(outfit, -1);
+						else
+							grand.PrintTrace("Invalid remove, outfit not found in existing illegals:");
+					}
+					else
+						illegals[GameData::Outfits().Get(grand.Token(0))] = grand.Value(1);
```

Comment:
`map::erase` when passed a key will attempt to erase the thing with that key if it exists and return 1 if it erases something or 0 if it doesn't.

Any particular reason an ignored outfit gets a value of -1? If you were later going to just use the `[]` operator to get the value, I'd understand since that'll give back 0 if the thing wasn't present, so you want -1 to be the value for a thing that is present but is to be ignored for fining. But, you're using `map::find` for that so I don't think it's necessary?
```suggestion
					if(grand.Token(0) == "remove" && !illegals.erase(GameData::Outfits().Get(grand.Token(1)))
						grand.PrintTrace("Invalid remove, outfit not found in existing illegals:");
					else if(grand.Token(0) == "ignore")
						illegals[GameData::Outfits().Get(grand.Token(1))] = 0;
					else
						illegals[GameData::Outfits().Get(grand.Token(0))] = grand.Value(1);
```

Reply:
Okay but now that means we're querying this map item 2 times, instead of just finding it once and modifying it if needs be.

You're right it looks like a negative value is not required.

The fine modification was happening at another level, but indeed I should modify government::Fines instead

Reply:
Maybe I've misunderstood, but I'm not sure where my version is querying the map twice.

Reply:
I'm the one who misunderstood actually, good suggestions!